<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Welcome to metaconv’s documentation! &mdash; metaconv 0.1 documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/sphinx_highlight.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="#" class="icon icon-home">
            metaconv
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <!-- Local TOC -->
              <div class="local-toc"><ul>
<li><a class="reference internal" href="#">Welcome to metaconv’s documentation!</a><ul>
<li><a class="reference internal" href="#format_functions.convert_oligos"><code class="docutils literal notranslate"><span class="pre">convert_oligos()</span></code></a></li>
<li><a class="reference internal" href="#format_functions.mothur_to_cutadapt"><code class="docutils literal notranslate"><span class="pre">mothur_to_cutadapt()</span></code></a></li>
<li><a class="reference internal" href="#manifest_create.read_folder_content"><code class="docutils literal notranslate"><span class="pre">read_folder_content()</span></code></a></li>
<li><a class="reference internal" href="#manifest_create.write_manifest"><code class="docutils literal notranslate"><span class="pre">write_manifest()</span></code></a></li>
</ul>
</li>
<li><a class="reference internal" href="#indices-and-tables">Indices and tables</a></li>
</ul>
</div>
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="#">metaconv</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="#" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Welcome to metaconv’s documentation!</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/index.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="welcome-to-metaconv-s-documentation">
<h1>Welcome to metaconv’s documentation!<a class="headerlink" href="#welcome-to-metaconv-s-documentation" title="Permalink to this heading"></a></h1>
<div class="toctree-wrapper compound">
</div>
<span class="target" id="module-format_functions"></span><dl class="py function">
<dt class="sig sig-object py" id="format_functions.convert_oligos">
<span class="sig-prename descclassname"><span class="pre">format_functions.</span></span><span class="sig-name descname"><span class="pre">convert_oligos</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">path</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'.'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">adapter_type</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'^'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">barcode_length</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">7</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#format_functions.convert_oligos" title="Permalink to this definition"></a></dt>
<dd><p>Converts the Mothur “oligo” format to .fasta files containing the barcodes, for use with Cutadapt.</p>
<p>The function reads the barcode pairs present in the oligo files with the .file extension used in the Mothur
software. Those barcode pairs are reformatted in two different .fasta files that can be used as inputs to
operate the demultiplexing of NGS reads with Cutadapt.
It outputs .fasta files in the directory where the oligo files are present.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>path</strong><span class="classifier">string</span></dt><dd><p>The path where the oligo files are stored, this directory will also be the output directory.
Default <cite>path = ‘.’</cite></p>
</dd>
<dt><strong>adapter_type</strong><span class="classifier">string</span></dt><dd><p>Specify the type of adapter for Cutadapt, according to the ones listed
<a class="reference external" href="https://cutadapt.readthedocs.io/en/stable/guide.html#adapter-types">here</a>.
Default <cite>adapter_type = ‘’</cite> that corresponds to an anchored 5’ adapter.</p>
</dd>
<dt><strong>barcode_length</strong><span class="classifier">integer</span></dt><dd><p>Specify the length of the barcodes in the oligo files.
Default <cite>barcode_length = 7</cite></p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="format_functions.mothur_to_cutadapt">
<span class="sig-prename descclassname"><span class="pre">format_functions.</span></span><span class="sig-name descname"><span class="pre">mothur_to_cutadapt</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">string_list</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">adapter_type</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'^'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">barcode_length</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">7</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#format_functions.mothur_to_cutadapt" title="Permalink to this definition"></a></dt>
<dd><p>Used from the “Convert Oligo” function to actually operate the conversion.
It reads line by line the previous file and reformats every line with a new string.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>string_list</strong></dt><dd><p>This parameter is created during the call of ‘Convert_oligos’.</p>
</dd>
<dt><strong>adapter_type</strong><span class="classifier">string</span></dt><dd><p>Specify the type of adapter for Cutadapt, according to the ones listed
<a class="reference external" href="https://cutadapt.readthedocs.io/en/stable/guide.html#adapter-types">here</a>.
Default <cite>adapter_type = ‘’</cite> that corresponds to an anchored 5’ adapter.</p>
</dd>
<dt><strong>barcode_length</strong><span class="classifier">integer</span></dt><dd><p>Specify the length of the barcodes in the oligo files.
Default <cite>barcode_length = 7</cite></p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

<span class="target" id="module-manifest_create"></span><dl class="py function">
<dt class="sig sig-object py" id="manifest_create.read_folder_content">
<span class="sig-prename descclassname"><span class="pre">manifest_create.</span></span><span class="sig-name descname"><span class="pre">read_folder_content</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">raw_files_path</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'.'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#manifest_create.read_folder_content" title="Permalink to this definition"></a></dt>
<dd><p>It reads the content of a folder to retrieve a list of <cite>.fastq.gz</cite> files.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>raw_files_path</strong></dt><dd><p>The path where the raw <cite>.fastq.gz</cite> files are stored.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="manifest_create.write_manifest">
<span class="sig-prename descclassname"><span class="pre">manifest_create.</span></span><span class="sig-name descname"><span class="pre">write_manifest</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">raw_files_path</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'.'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#manifest_create.write_manifest" title="Permalink to this definition"></a></dt>
<dd><p>It writes a manifest file to use as import with Qiime2 from a folder of ´.fastq.gz’files.
The content of the folder is read and all the ´.gz´ files included in the manifest.
The manifest file can then be used normally with Qiime2 to start the analysis pipeline.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>raw_files_path</strong></dt><dd><p>The path where the raw <cite>.fastq.gz</cite> files are stored.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="indices-and-tables">
<h1>Indices and tables<a class="headerlink" href="#indices-and-tables" title="Permalink to this heading"></a></h1>
<ul class="simple">
<li><p><a class="reference internal" href="genindex.html"><span class="std std-ref">Index</span></a></p></li>
<li><p><a class="reference internal" href="py-modindex.html"><span class="std std-ref">Module Index</span></a></p></li>
<li><p><a class="reference internal" href="search.html"><span class="std std-ref">Search Page</span></a></p></li>
</ul>
</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023, Lorenzo Assentato.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>